make-prs.py

```diff
@@ -0,0 +1,251 @@
+import argparse
+import logging
+import os
+import re
+import shutil
+import subprocess
+import typing
+
+import github
+import requests
+
+logger = logging.getLogger(__name__)
+logging.basicConfig(level=logging.INFO)
+
+WORKING_DIR = "./tmp"
+DIFF_FILE = "./diff.patch"
+
+auth = github.Auth.Token(os.getenv("GITHUB_TOKEN"))
+gh_client = github.Github(auth=auth)
+
+
+def get_diff_as_patch(repo: str, pull_request_number: int):
+    logger.info("Getting diff from PR ...")
+
+    pr_diff_url = gh_client.get_repo(repo).get_pull(pull_request_number).diff_url
+
+    res = requests.get(pr_diff_url)
+    if res.status_code != requests.codes.ok:
+        logger.error(f"Error getting diff from PR {pull_request_number}.")
+        exit(1)
+
+    with open(DIFF_FILE, "w") as f:
+        f.write(res.text)
```

Comment:
Is the resulting patch file annotated with commit parents?

Reply:
No, only two commit IDs. Example:

```
diff --git a/src/charm.py b/src/charm.py
index 1851143..778eecb 100755
--- a/src/charm.py
+++ b/src/charm.py
...
```